### PR TITLE
Fix 3.9 branch commits and add more Katello 4.11 GA info

### PR DESCRIPTION
**Pull request by ianballou**
I noticed that numerous commits have gone missing from the 3.9 branch of foreman-documentation.

image

I think https://github.com/theforeman/foreman-documentation/commit/d9e7e8c438bc809773880430d4ec3d10b6703fbb might've been added via a force-push that cleared out the history of prior commits.

In a snippet from https://api.github.com/repos/theforeman/foreman-documentation/events, I saw this:

```
...
    "payload": {
      ...
      "ref": "refs/heads/3.9",
      "head": "d9e7e8c438bc809773880430d4ec3d10b6703fbb",
      "before": "c677a3a0777145c295921f6b3b48fab95a18d496",
      "commits": [
        {
          "sha": "d9e7e8c438bc809773880430d4ec3d10b6703fbb",
          ...
          "message": "Update DocState to stable",
          ...
        }
      ]
    },
...
```

If you look at the the "before" hash (https://github.com/theforeman/foreman-documentation/commits/c677a3a0777145c295921f6b3b48fab95a18d496), it contains all the commits that the 3.9 branch should have.

If you look at the "head" hash (https://github.com/theforeman/foreman-documentation/commits/d9e7e8c438bc809773880430d4ec3d10b6703fbb), it contains only https://github.com/theforeman/foreman-documentation/commit/d9e7e8c438bc809773880430d4ec3d10b6703fbb  and then some commits that are a bit old at this point.

Regardless, I've reconstructed what should be the state of the 3.9 branch before the force-push. I happened to have a local copy that I was working on. I also cherry-picked in one commit that came in after my change.

---

I've also included my Katello 4.11 GA commit in the mix that updates the changelogs as the very last commit.

### CODE REVIEW

**Comment by maximiliankolb** [comment]
**TODO**: Merged but not cherry-picked:

* https://github.com/theforeman/foreman-documentation/pull/2608
* https://github.com/theforeman/foreman-documentation/pull/2645

I can take care of this after the "3.9" state has been restored.

As of this morning, my local repo looks as follows:
````
$ git status
On branch 3.9
Your branch and 'upstream/3.9' have diverged,
and have 17 and 1 different commits each, respectively.
  (use "git pull" to merge the remote branch into yours)

nothing to commit, working tree clean
$ git log -n 1
commit 5afac4cd9cd1cca6d1b87a0e690dde8bfba2be79 (HEAD -> 3.9)
Author: Brian Angelica <91690569+bangelic@users.noreply.github.com>
Date:   Wed Dec 13 03:13:10 2023 -0500

    Add lifecycle to path in navigation (#2635)
    
    (cherry picked from commit 6968f71b4c7634ab0d2bb44cdee8dba4b9f3ce6a)
````

**Review by maximiliankolb — APPROVED** [review]
I had a look at your PR locally and compared your branch to my local "3.9" branch:

````
$ git fetch upstream pull/2644/head:review_2644
$ git diff --name-only 3.9 review_2644
guides/common/attributes.adoc                                                                                                                                                                                 
guides/common/modules/con_pull-based-rex-transport-tuning.adoc
guides/common/modules/proc_decreasing-performance-impact-of-the-pull-based-rex-transport.adoc
guides/common/modules/proc_disabling-puppet.adoc
guides/common/modules/proc_increasing-host-limit-for-pull-based-rex-transport.adoc
guides/common/modules/ref_system-requirements-for-tuning.adoc
guides/doc-Release_Notes/master.adoc
guides/doc-Release_Notes/topics/foreman-3.9.0.adoc
guides/doc-Release_Notes/topics/foreman-contributors.adoc
guides/doc-Release_Notes/topics/foreman.adoc
guides/doc-Release_Notes/topics/katello-4.11.0.adoc
guides/doc-Release_Notes/topics/katello.adoc
guides/doc-Tuning_Performance/master.adoc
````

These are either files I know we've been working on recently (e.g. REX Pull; performance tuning; disabling Puppet), or are Release Notes-related. :+1: 

I did not review the diff closely, but I think this looks good. Thank you Ian for investigating and providing a fix!

**Comment by mjivraja** [comment]
Out of curiosity, did this happen because the feature branch wasn't properly rebased with the source branch?

**Comment by mjivraja** [comment]
> Not a rebasing problem -- the release owner sometimes pushes code to release branches directly, but in this case it was an accidental force push from a local directory that was missing commits. The force push overwrote the commit history, which made it look like commits went missing. 
> 
> Thankfully even with a force push, things aren't completely erased from git!

Oww yes, I saw `master` in the PR that erased commits, which I assumed to be the same name feature branch initially. The author might have missed pulling the latest changes from the remote branch.

Thank you for your explanation 👍

**Comment by Griffin-Sullivan** [comment]
I feel like there's other steps that have you push directly to non-master branches, but I'm not really sure why I chose to push directly instead of make a PR. We could add a note in the procedure.

**Comment by maximiliankolb** [comment]
We should configure the project to not allow force pushes to "X.Y" & "master" branches.

File: guides/doc-Release_Notes/topics/katello.adoc [hunk 1]
```diff
@@ -41,7 +41,7 @@ Features performance improvements most notably around import/export for disconne
 
 === Pulpcore
 
-The pulpcore module stream must be disabled before updating packages. 
+The pulpcore module must be disabled before upgrading to Katello 4.11.
```
Comment (ekohl) [inline guides/doc-Release_Notes/topics/katello.adoc:44]: Note to self: there is a step in the upgrade guide that https://github.com/theforeman/foreman-documentation/pull/2644/commits/43e1e32c905af403ca464b5ac90c2c6081567e76 adds again.

**Review by ekohl — APPROVED** [review]
Let's get this in.

**Pull request merged**